### ISSUE TITLE
Players can use the `InventoryProfilesNext` mod to make the gun have bullets beyond the limitation.

### ISSUE THREAD

**Issue by freeze-dolphin** [WeaponMechanics/MechanicsMain#351]
### Current Behavior

With [`InventoryProfilesNext`](https://www.curseforge.com/minecraft/mc-mods/inventory-profiles-next), player can switch between two identical items in the backpack and in the hand by using the `Auto Refill` feature.

By taking advantage of this feature, player can get guns with ammo loaded whose number is beyond the limitation of the gun. 

Sorry for my broken English. XD

### Expected Behavior

The ammo of a gun should never exceed the limitation of itself.

### Steps To Reproduce

It's hard to describe precisely, but there is a video from one player in my server.

0. Have `IPN` installed, configure the mod like this (in `Auto Refill` page):
   ![image](https://github.com/WeaponMechanics/MechanicsMain/assets/40729044/f90c6e19-a031-4a88-9414-c292180d60d3)
   ![image](https://github.com/WeaponMechanics/MechanicsMain/assets/40729044/89a9bfad-74e1-4649-be1a-548b105aab1b)
1. Get two identical weapon. (default AK_47 in video)
2. Clear the clip of one weapon (`A`), and keep the other's full (`B`). Then hold `A` and put `B` into inventory
3. Hold `A` and reload (using right click), then drop it before reloading is done. `IPN` will auto refill `B` into your hand
4. Wait until reloading is done, and now you get an AK_47 with 60 bullets in clip

video is below

### Version and Log

no error log

version:

```
> wm info
[11:29:53 INFO]: WeaponMechanics, v2.6.5
[11:29:53 INFO]:   ➢ Authors: DeeCaaD, CJCrafter
[11:29:53 INFO]:   ➢ Command: /weaponmechanics
[11:29:53 INFO]:   ➢ Server: Paper git-Paper-131 (MC: 1.20.1)
[11:29:53 INFO]:   ➢ MechanicsCore: 2.4.11
[11:29:53 INFO]:   ➢ Java: 17.0.8
[11:29:53 INFO]:   ➢ Supported plugins: No supported plugins installed
```

### Anything else?

video:

https://github.com/WeaponMechanics/MechanicsMain/assets/40729044/a96fd12e-2f79-4b77-aac2-00b01928639e

**Comment by CJCrafter** [comment]
Using a modded client makes it possible to cheat, there are many more ways to cheat past this

**Comment by freeze-dolphin** [comment]
`IPN` is a widely used mod, and the player in my server found the issue somewhat 'accidentally' during his normal usage.

I think there should be a checker for switching weapon while reloading.

If it is detected that the player switched the item during reloading, the reloading should be canceled.

**Comment by CJCrafter** [comment]
We do cancel reload when you switch items. You can test this by switching items normally while reloading. Very little I can do except suggest you use an anti-cheat.

**Comment by CJCrafter** [comment]
I unfortunately do not have the time to download mods and determine how they break MC code.

**Comment by XyBv587** [comment]
When reloading the AK47, opening the backpack, then pointing the mouse to another AK47, and using the number keys to switch between the AK47 in the backpack and the quick access bar, a bug occurs. No mods are required.

**Comment by CJCrafter** [comment]
Fixed in #458
